src/adapter/src/coord/consistency.rs

```diff
@@ -71,34 +71,34 @@ impl Coordinator {
 
     /// # Invariants:
     ///
-    /// * Read capabilities should reference known objects.
+    /// * Read holds should reference known objects.
     ///
-    fn check_read_capabilities(&self) -> Result<(), Vec<ReadCapabilitiesInconsistency>> {
-        let mut read_capabilities_inconsistencies = Vec::new();
-        for (gid, _) in &self.storage_read_capabilities {
-            if self.catalog().try_get_entry(gid).is_none() {
-                read_capabilities_inconsistencies
-                    .push(ReadCapabilitiesInconsistency::Storage(gid.clone()));
+    fn check_read_holds(&self) -> Result<(), Vec<ReadHoldsInconsistency>> {
+        let mut inconsistencies = Vec::new();
+
+        for timeline in self.global_timelines.values() {
+            for id in timeline.read_holds.storage_ids() {
+                if self.catalog().try_get_entry(&id).is_none() {
+                    inconsistencies.push(ReadHoldsInconsistency::Storage(id));
+                }
             }
-        }
-        for (gid, _) in &self.compute_read_capabilities {
-            if !gid.is_transient() && self.catalog().try_get_entry(gid).is_none() {
-                read_capabilities_inconsistencies
-                    .push(ReadCapabilitiesInconsistency::Compute(gid.clone()));
+            for (_, id) in timeline.read_holds.compute_ids() {
```

Comment:
We probably want to check the cluster ID here too.

Reply:
Added a check for the cluster ID.